extension/public/background.ts

```diff
@@ -2,9 +2,17 @@ import {
   initContentScriptMessageListener,
   initExtensionMessageListener,
   initInstalledListener,
+  initInitAlarmListener,
 } from "background";
 
-initContentScriptMessageListener();
-initExtensionMessageListener();
+import { buildStore } from "background/store";
 
-initInstalledListener();
+async function main() {
+  const store = await buildStore();
```

Comment:
We need to build the store async and at the top level to make sure that in v2 mode, it is only built when the background script is ran and to use the same hydrated store across listeners in v3 mode.